config/value.go

```diff
@@ -333,10 +333,41 @@ func derefType(t reflect.Type) reflect.Type {
 	return t
 }
 
+func convertPrimitiveValue(value interface{}, targetType reflect.Type) (interface{}, error) {
+	ret, err := convertVal(value, targetType)
+	if ret == nil {
```

Comment:
Not sure if we are talking about the same thing. I just mean:
if err != nil {
return nil, fmt.Errorf("can't convert %v to %v", reflect.TypeOf(value).String(), targetType)
}
return ret, nil

i.e. the if condition should be on err and not ret